.github/workflows/cuda-gpu-ci.yml

```diff
@@ -1,31 +1,33 @@
 name: CUDA GPU
+
+# Only run this workflow when a Pull Request is labeled with the
+# 'CUDA CI' label.
 on:
-  workflow_dispatch:
-    inputs:
-      commit_hash:
-        description: Commit hash to test
-        required: true
+  pull_request:
+    types:
+      - labeled
 
-permissions: read-all
+# In order to remove the "CUDA CI" label we need to have write permissions for PRs
+permissions:
+  pull-requests: write
 
 jobs:
   tests:
+    if: contains(github.event.pull_request.labels.*.name, 'CUDA CI')
     runs-on:
       group: cuda-gpu-runner-group
     name: Run Array API unit tests
     steps:
+      - uses: actions-ecosystem/action-remove-labels@v1.3.0
```

Comment:
```suggestion
      - uses: actions-ecosystem/action-remove-labels@v1
```

I think the policy is to use major number versions as discussed previously: https://github.com/scikit-learn/scikit-learn/pull/29203#issuecomment-2154372437.